test/ability_flower_gift.c

```diff
@@ -0,0 +1,86 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim in harsh sunlight")
+{
+    GIVEN {
+        PLAYER(SPECIES_CHERRIM) { Ability(ABILITY_FLOWER_GIFT); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUNNY_DAY); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim back to normal when weather changes")
+{
+    GIVEN {
+        PLAYER(SPECIES_CHERRIM) { Ability(ABILITY_FLOWER_GIFT); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUNNY_DAY); }
+        TURN { MOVE(opponent, MOVE_RAIN_DANCE); }
+    } SCENE {
+        // transforms in sun
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+        // back to normal
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim back to normal when its ability is suppressed")
+{
+    GIVEN {
```

Comment:
Needs an assumption for `B_WEATHER_FORM_SUPPRESS >= GEN_5`.

Reply:
Still needs the assumption (though for `B_WEATHER_FORM` instead now), otherwise it'll cause this to happen when switched:
```
Flower Gift transforms Cherrim back to normal when its ability is suppressed: FAIL
test/ability_flower_gift.c:48: Unmatched ABILITY_POPUP
Flower Gift transforms Cherrim in harsh sunlight: FAIL
test/ability_flower_gift.c:12: Unmatched ABILITY_POPUP
Flower Gift increases the Sp. Def of Cherrim and its allies by 1.5x 2/2: FAIL
test/ability_flower_gift.c:108: Unmatched ABILITY_POPUP
Flower Gift increases the attack of Cherrim and its allies by 1.5x 2/2: FAIL
test/ability_flower_gift.c:75: Unmatched ABILITY_POPUP
Flower Gift transforms Cherrim back to normal when weather changes: FAIL
test/ability_flower_gift.c:28: Unmatched ABILITY_POPUP
Forecast transforms Castform back to normal when its ability is suppressed: FAIL
test/ability_forecast.c:251: Unmatched ANIMATION
```